hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java

```diff
@@ -23,60 +23,95 @@
 import com.hazelcast.internal.serialization.SerializationService;
 import com.hazelcast.jet.Job;
 import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.core.JobNotFoundException;
+import com.hazelcast.jet.core.JobStatus;
 import com.hazelcast.jet.impl.util.NonCompletableFuture;
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.logging.LoggingService;
 import com.hazelcast.spi.exception.TargetDisconnectedException;
 import com.hazelcast.spi.exception.TargetNotMemberException;
 
 import javax.annotation.Nonnull;
-import java.util.UUID;
 import java.util.concurrent.CancellationException;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.LockSupport;
 import java.util.function.BiConsumer;
 import java.util.function.Supplier;
 
 import static com.hazelcast.jet.impl.util.ExceptionUtil.peel;
 import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
+import static com.hazelcast.jet.impl.util.ExceptionUtil.withTryCatch;
 import static com.hazelcast.jet.impl.util.Util.memoizeConcurrent;
 import static com.hazelcast.jet.impl.util.Util.toLocalDateTime;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 /**
  * Base {@link Job} implementation for both client and member proxy.
  */
-public abstract class AbstractJobProxy<T> implements Job {
+public abstract class AbstractJobProxy<ContainerType, MemberIdType> implements Job {
```

Comment:
The fact that these generic types don't consist of single letters creates a bit of confusion, especially when reading from a place that doesn't have any coloring. At first, I though it was a class name.